overlord/configstate/configcore/netplan.go

```diff
@@ -92,18 +117,137 @@ func getNetplanFromSystem(key string) (result interface{}, err error) {
 	netplan := conn.Object("io.netplan.Netplan", "/io/netplan/Netplan")
 
 	if err := netplan.Call("io.netplan.Netplan.Config", 0).Store(&netplanConfigSnapshotBusAddr); err != nil {
-		if derr, ok := err.(dbus.Error); ok {
-			// Having no netplan config is *not* an error, we just
-			// do not support netplan config.
-			if isNoServiceOrMethodErr(derr) {
-				return nil, nil
-			}
+		return nil, err
+	}
+	logger.Debugf("using netplan config %v", netplanConfigSnapshotBusAddr)
+
+	netplanCfgSnapshot := conn.Object("io.netplan.Netplan", dbus.ObjectPath(netplanConfigSnapshotBusAddr))
+	return netplanCfgSnapshot, nil
+}
+
+func validateNetplanSettings(tr config.Conf) error {
+	// XXX: validate somehow once we support writing?
+	return nil
+}
+
+func hasNetplanChanges(tr config.Conf) bool {
+	for _, chg := range tr.Changes() {
+		if chg == "core.system.network.netplan" || strings.HasPrefix(chg, "core.system.network.netplan.") {
+			return true
+		}
+	}
+	return false
+}
+
+func handleNetplanConfiguration(tr config.Conf, opts *fsOnlyContext) error {
+	if !hasNetplanChanges(tr) {
+		return nil
+	}
+
+	var cfg map[string]interface{}
+	if err := tr.Get("core", "system.network.netplan", &cfg); err != nil && !config.IsNoOption(err) {
```

Comment:
is `IsNoOption` possible if we have confirmed that there is a netplan related change before?